src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
     val toResizeArrayAsync: source: taskSeq<'T> -> Task<ResizeArray<'T>>
 
-    /// Unwraps the taskSeq as a Task<IList<_>>. This function is non-blocking.
+    /// <summary>
+    /// Builds an <see cref="IList&lt;'T>" /> asynchronously from the input task sequence in <paramref name="source" />.
+    /// This function is non-blocking while it builds the IList.
+    /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting IList interface.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toIListAsync: source: taskSeq<'T> -> Task<IList<'T>>
 
-    /// Create a taskSeq of an array.
+    /// <summary>
+    /// Views the given <see cref="array" /> as a task sequence, that is, as an <see cref="IAsyncEnumerable&lt;'T>" />.
+    /// </summary>
+    ///
+    /// <param name="source">The input array.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input array is null.</exception>
     val ofArray: source: 'T[] -> taskSeq<'T>
 
-    /// Create a taskSeq of a list.
+    /// <summary>
+    /// Views the given <see cref="list" /> as a task sequence, that is, as an <see cref="IAsyncEnumerable&lt;'T>" />.
+    /// </summary>
+    ///
+    /// <param name="source">The input list.</param>
+    /// <returns>The resulting task sequence.</returns>
     val ofList: source: 'T list -> taskSeq<'T>
 
-    /// Create a taskSeq of a seq.
+    /// <summary>
+    /// Views the given <see cref="seq" /> as a task sequence, that is, as an <see cref="IAsyncEnumerable&lt;'T>" />.
+    /// </summary>
+    ///
+    /// <param name="source">The input sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val ofSeq: source: seq<'T> -> taskSeq<'T>
 
-    /// Create a taskSeq of a ResizeArray, aka List.
+    /// <summary>
+    /// Views the given resizable array as a task sequence, that is, as an <see cref="IAsyncEnumerable&lt;'T>" />.
+    /// </summary>
+    ///
+    /// <param name="source">The input resize array.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input resize array is null.</exception>
     val ofResizeArray: source: ResizeArray<'T> -> taskSeq<'T>
 
-    /// Create a taskSeq of a sequence of tasks, that may already have hot-started.
+    /// <summary>
+    /// Views the given <see cref="seq" /> of <see cref="Task" />s as a task sequence, that is, as an
+    /// <see cref="IAsyncEnumerable&lt;'T>" />. A sequence of tasks is not the same as a task sequence.
+    /// Each task in a sequence of tasks can be run individually and potentially out of order, or with
```

Comment:
great to have this laid out
in practice does the fact that most/all tasks will be started not mean that the dominant consideration will be the unconstrained overlapping execution?
I'd focus on the ordered waiting as the thing being delivered vs any unsupportable/not explictly achieved limitation on overlapping afforded by this?
... or I might have it wrong (for instance I don't know the typical ways in which you arrive at a task that's not already started)
Ah, I get it now - you are trying to convey that the evauation is lazy and that this avoids the overlapping execution of the Task
Not sure if anyone else will follow the same thought process, as it really does depend on where you're at in grokking it all (I have some residual mental mappings where I picture an AsyncSeq yielding materialized spans of items that get `collect`ed into a flat sequence which are definitely not helpful)

Reply:
Also the massive amount of work and value in this PR is the docs work here - the title should probably reflect that, not the fact that there happens to be a feature request ebing fulfilled too!

Reply:
A list/seq/array of tasks may or may not be hot-started. It is a bad design decision to do this. In fact, I have tests that show exactly why this is a bad idea _in principle_. However, before `IAsyncEnumerable` this was probably the dominant way, hence providing such a function now (including the warning).

Leaving as is for now unless I can come up with better wording.